Goutte/Client.php

```diff
@@ -62,6 +62,11 @@ public function removeHeader($name)
         unset($this->headers[$name]);
     }
 
+    public function resetHeaders()
```

Comment:
I would prefer to have a `reset()` method that resets all settings, including auth for instance.